tensorboard/webapp/routes/dashboard_deeplink_provider.ts

```diff
@@ -132,6 +158,22 @@ export class DashboardDeepLinkProvider extends DeepLinkProvider {
         case SMOOTHING_KEY:
           smoothing = Number(value);
           break;
+        case COLOR_GROUP_KEY: {
+          switch (value) {
+            case 'experiment':
+              groupBy = {key: GroupByKey.EXPERIMENT};
+              break;
+            case 'run':
+              groupBy = {key: GroupByKey.RUN};
+              break;
+          }
+
+          if (value.startsWith('regex:')) {
+            const [, regexString] = value.split(':', 2);
```

Comment:
if the regex is `foo\:bar`, the value would be `regex:foo\:bar`
but the regexString here becomes `foo`
why adding `regex:` prefix then extract here?

Reply:
Ah, I misunderstood `split` API (thought it would behave like Python's `split`). 

> why adding regex: prefix then extract here?

We want to be able to distinguish different modes with the same query parameter, `?colorBy`. For example, 
```
?colorBy=experiment -> {key: GroupByKey.EXPERIMENT}
?colorBy=regex:<string> -> {key: GroupByKey.REGEX, regexString: <string>}
```